src/main/java/org/opensearch/commons/InjectSecurity.java

```diff
@@ -143,13 +144,16 @@ public void injectUserInfo(final User user) {
                 );
             return;
         }
+
         StringJoiner joiner = new StringJoiner("|");
```

Comment:
Curious to know where this serialization logic is used because I thought security was the only place where the user object would be serialized into the ThreadContext. Is this used for interplay between plugins in certain contexts?

Reply:
While it may make sense to centralize this logic to prevent duplication, long term I'd like to remove a lot of security logic from common-utils including this.

Reply:
> Curious to know where this serialization logic is used because I thought security was the only place where the user object would be serialized into the ThreadContext. Is this used for interplay between plugins in certain contexts?

Yes, this is used for plugins to communicate with one another.



> While it may make sense to centralize this logic to prevent duplication, long term I'd like to remove a lot of security logic from common-utils including this.

I agree, this is really needed, but that might require some security features in core.

Reply:
I have an RFC open in the security repo that outlines 4 security features that need to be built in order for plugins to migrate off of using common-utils: https://github.com/opensearch-project/security/issues/5052

First, the security repo needs to support existing use cases in a model that's opt-in for plugins so that they can adopt the new mechanisms and create migration paths. Particularly, a lot of plugins store a copy of a user in their own system indices and eventually I think we should remove the copies but mechanisms need to be built to support it first.